python/packages/nisar/products/writers/BaseL2WriterSingleInput.py

```diff
@@ -1506,6 +1507,70 @@ def populate_processing_information_l2_common(self):
                 output_ds_name_list=ds_name_list,
                 skip_if_not_present=True)
 
+    def geocode_isce3_lut(self, correction_lut, lut_name,
+                          timing_corrections_group_path, frequency):
+        '''
+        Geocode ISCE3 look-up table (LUT) object
+
+        Parameters
+        ----------
+        correction_lut: isce3.core.LUT2d
+            ISCE3 look-up table (LUT) object containing the data
+            to be geocoded
+        lut_name: str
+           Name of the LUT dataset in the output product metadata
+        timing_corrections_group_path: str
+            Path to the output HDF5 LUT group
+        frequency: str, optional
+            Frequency sub-band, used to read the sub-band wavelength.
+        '''
+
+        new_var_array = correction_lut.data
+
+        driver = gdal.GetDriverByName("GTiff")
+        dtype = gdal_array.NumericTypeCodeToGDALTypeCode(
+                new_var_array.dtype)
+        length, width = new_var_array.shape
+
+        scratch_path = self.cfg['product_path_group']['scratch_path']
+        temp_file = tempfile.NamedTemporaryFile(dir=scratch_path,
+                                                suffix='.tif')
+        dset = driver.Create(temp_file.name, xsize=width, ysize=length,
+                             bands=1, eType=dtype)
+        raster_band = dset.GetRasterBand(1)
+        raster_band.WriteArray(new_var_array)
+
+        # flush data to the disk (gc is the garbage collector)
+        raster_band.FlushCache()
+        del raster_band
+        del dset
+        gc.collect()
+
+        correction_raster = isce3.io.Raster(temp_file.name)
```

Comment:
It's no longer necessary to create the raster using GDAL, write contents to it, and then re-open it using `isce3.io.Raster`. We recently added the ability to write data to `Raster` objects using isce3 in Python, so you should be able to simply create the raster object using isce3 and then write to it.

7b9285ee86f62647b3a736c0b00b542e90882f70